src/test/groovy/de/dkfz/roddy/tools/RoddyIOHelperMethodsSpecification.groovy

```diff
@@ -0,0 +1,131 @@
+package de.dkfz.roddy.tools
+
+import spock.lang.Shared
+import spock.lang.Specification
+
+import java.awt.image.ImagingOpException
+
+import static de.dkfz.roddy.tools.RoddyIOHelperMethods.NativeLinuxZipCompressor
+
+class RoddyIOHelperMethodsSpecification extends Specification {
+
+    @Shared
+    def zipCompressor = new NativeLinuxZipCompressor()
+
+    def "FindComponentIndexInPath"() {
+        when:
+        def result = RoddyIOHelperMethods.findComponentIndexInPath("/a/b/\${sample}/d", '${sample}')
+        then:
+        result == Optional.of(2)
+    }
+
+    def "test get native linux zip compressor compression string"(String source, String zipFile, String result) {
+        expect:
+        zipCompressor.getCompressionString(new File(source), new File(zipFile)).toString() == result
+
+        where:
+        source        | zipFile            | result
+        "/source/abc" | "/target/test.zip" | '[[ -f "/target/test.zip" ]] && rm /target/test.zip; cd /source && zip -r9 /target/test.zip abc > /dev/null && md5sum /target/test.zip'
+    }
+
+    def "test get native linux zip compressor decompression string"(String from, String to, String result) {
+        expect:
+        zipCompressor.getDecompressionString(new File(from), new File(to)).toString() == result
+
+        where:
+        from              | to                | result
+        "/source/abc.zip" | "/target/results" | "[[ ! -d /target/results ]] && mkdir -p /target/results; cd /target/results && unzip -o /source/abc.zip > /dev/null"
+    }
+
+    def "Test correct run of native linux zip compression and decompression on directory. The test shall return without exceptions!"() {
+        setup: "First setup input data"
+
+        def rangeOfFiles = ["a", "b", "c"]
+        File tempFolder = File.createTempDir()
+        tempFolder.deleteOnExit()
```

Comment:
Did you check, whether this is deleted even if files are in there? We had this problem with tons of file before and I am not 100 sure where they came from. There is also a TemporaryFolder @Rule in junit4.

Reply:
Nope did not. But I changed to a @Rule with TemporaryFolder